src/LivewireRelay.php

```diff
@@ -4,32 +4,27 @@
 
 use Livewire\Component;
 use Livewire\Features\SupportEvents\Event;
-use Livewire\LivewireManager;
-use Livewire\Mechanisms\DataStore;
+use Livewire\Livewire;
 use Livewire\Mechanisms\HandleComponents\ComponentContext;
 
+use function Livewire\store;
+
 /** @internal */
 final readonly class LivewireRelay
 {
     public const EVENT = 'toaster:received';
 
-    public function __construct(
-        private DataStore $store,
-        private LivewireManager $livewire,
-        private Collector $toasts,
-    ) {}
-
     public function __invoke(Component $component, ComponentContext $ctx): void
     {
-        if (! $this->livewire->isLivewireRequest()) {
+        if (! Livewire::isLivewireRequest()) {
```

Comment:
Basically I prefer using dependencies, since I'm not sure how SSL store is global context between requests. We really do not want to release toast for user that doesn't perform specific action. But if there a global context for each request - it may be a problem. 

There is not an easy way to check it on production :)

Reply:
There is no global context. SSL means "Static Service Location" which is the technique being used when using e.g. `Facade` classes. As I understood it the problem lies in the fact that an old service binding is kept in-memory which is no longer being used, resulting in malfunctions because the toasts are sent to an old instance. This happens because `LivewireManager` is registered as a `singleton` and not `scoped`. So the event listener has to pull in fresh dependencies each time and by using `Facade` classes we can ensure this because `Octane` resets the `Facade` binding resolution after each requests AFAIK.